physics/cu_unified_deep.F90

```diff
@@ -0,0 +1,6052 @@
+    real(kind=kind_phys), dimension (its:ite,kts:kte) :: drag, buo, zi, del
+    real(kind=kind_phys), dimension (its:ite,kts:kte),intent (out) :: wu2,omega_u,      &
+         zeta,zdqca
+    real(kind=kind_phys), dimension (its:ite),intent(out) :: wc,omegac
+    real(kind=kind_phys) :: rho,bb1,bb2,dz,dp,ptem,tem1,ptem1,tem,rfact,gamma,val
+    integer :: i,k    
+
+  
+    !  compute updraft velocity square(wu2)
+    !> - Calculate updraft velocity square(wu2) according to Han et al.'s (2017) \cite han_et_al_2017 equation 7.
+    !LB: This routine outputs updraft velocity square (m/s), updraft omega_u (Pa/s), and cloud average updraft 
+    !velocity (m/s) and omega_u (Pa/s) in the case progsima is true.
+    
+    do k = 1, ktf
+       do i = 1,itf
+          wu2(i,k)=0.
+          drag(i,k)=0.
+          buo(i,k)=0.
+          omega_u(i,k)=0.
+          zeta(i,k)=0.
+          zdqca(i,k)=0.
+       enddo
+    enddo
+
+    do i=1,itf
+       wc(i)=0.
+       omegac(i)=0.
+       sumx(i)=0.
+    enddo
+
+    do k = 1, ktf-1
+       do i = 1,itf
+          zi(i,k) = 0.5*(zo(i,k)+zo(i,k+1))
+          del(i,k) = delp(i,k)*0.001
+       enddo
+    enddo
+    
+    do k = 2, ktf-1
+       do i = 1, itf
+         if (ierr(i)==0) then
+           if(k >= kbcon(i) .and. k < ktcon(i))then
+              gamma   = el2orc * qeso(i,k+1) / (to(i,k+1)**2)
+              if(k >= kbcon(i) .and. clw_all(i,k)>0.)then
+               buo(i,k) = buo(i,k) - g * qlk(i,k)  
+              endif      
+              rfact =  1. + fv * cp * gamma * to(i,k) / xlv
+              buo(i,k) = buo(i,k) + (g / (cp * to(i,k))) * dbyo(i,k) / (1. + gamma) * rfact
+              val = 0.
+              buo(i,k) = buo(i,k) + g * fv * max(val,(qeso(i,k) - qo(i,k)))
+              buo(i,k) = max(val,buo(i,k))
+              drag(i,k) = max(entr_rate_2d(i,k),cd(i,k))
+           endif
+        endif
+       enddo
+    enddo
+
+      bb1 = 4.0
+      bb2 = 0.8
+      do k = 2, ktf-1
+        do i = 1, itf
+          if (ierr(i)==0) then
+            if(k > kbcon(i) .and. k < ktcon(i)) then
+              dz    = zi(i,k) - zi(i,k-1)
+              tem  = 0.25 * bb1 * (drag(i,k)+drag(i,k-1)) * dz
+              tem1 = 0.5 * bb2 * (buo(i,k)+buo(i,k-1)) * dz
+              ptem = (1. - tem) * wu2(i,k-1)
+              ptem1 = 1. + tem
+              wu2(i,k) = (ptem + tem1) / ptem1
+              wu2(i,k) = max(wu2(i,k), 0.)
+            endif
+          endif
+        enddo
+      enddo
+
+      if(progsigma)then                                                                                                                    
+          do k = 2, ktf-1
+            do i = 1, itf
+               if (ierr(i)==0) then
+                  if(k > kbcon(i) .and. k < ktcon(i)) then
+                     rho = po(i,k)*100. / (rd * to(i,k))
+                     omega_u(i,k)=-1.0*sqrt(wu2(i,k))*rho*g
+                     omega_u(i,k)=MAX(omega_u(i,k),-80.)
```

Comment:
If the limit (-80) on `omega_u` might be changeable/tunable in the future, consider adding a variable/parameter for it and setting it at the top of the subroutine.